.github/workflows/qodana_code_quality.yml

```
@@ -0,0 +1,53 @@
+name: Qodana
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+jobs:
+  qodana:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)